Source/Core/InputCommon/ControllerInterface/evdev/evdev.cpp

```diff
@@ -44,6 +45,30 @@ class InputBackend final : public ciface::InputBackend
   void StopHotplugThread();
   void HotplugThreadFunc();
 
+  // Linux has the strange behavior that closing file descriptors of event devices can be
+  // surprisingly slow, in the range of 20-70 milliseconds. For modern systems that have maybe 30
+  // event devices this can quickly add up, leading to visibly slow startup. So we close FDs on a
+  // separate thread *shrug*
+  friend evdevDevice;
+  void DeferredClose(int fd) { m_cleanup_queue.emplace_back(fd); }
+  void CleanUp()
+  {
+    if (m_cleanup_queue.empty())
+      return;
+    std::vector<int> work;
+    m_cleanup_queue.swap(work);
+    std::thread(
+        [](std::vector<int> vec) {
+          Common::SetCurrentThreadName("evdev cleanup");
+          for (int fd : vec)
+            close(fd);
+        },
+        std::move(work))
+        .detach();
```

Comment:
`detach`ing threads is ugly. You could replace this with our `WorkQueueThread`.

Reply:
Good idea, done.